great_expectations/core/usage_statistics/anonymizers/profiler_run_anonymizer.py

```diff
@@ -172,9 +172,18 @@ def _anonymize_domain_builder(self, domain_builder: dict) -> dict:
 
         batch_request: Optional[dict] = domain_builder.get("batch_request")
         if batch_request:
+            anonymized_batch_request: Optional[Union[str, dict]]
+            if isinstance(batch_request, str):
+                anonymized_batch_request = self.anonymize(batch_request)
+            else:
+                anonymized_batch_request = (
+                    self._batch_request_anonymizer.anonymize_batch_request(
+                        **batch_request
+                    )
+                )
```

Comment:
Due to not performing variable substitution at this point, we must account for the case where the batch request is a `$variable` string.